extension/src/experiments/data/index.ts

```diff
@@ -86,6 +93,58 @@ export class ExperimentsData extends BaseData<ExpShowOutput> {
     this.collectedFiles = collectFiles(data, this.collectedFiles)
   }
 
+  private async collectRevisionGitDetails(
+    branch: string,
+    hashes: Record<string, HashInfo>,
+    flags: (ExperimentFlag | string)[]
+  ) {
+    const nbOfCommitsToShow = this.experiments.getNbOfCommitsToShow(branch)
+
+    for (let i = 0; i < nbOfCommitsToShow; i++) {
+      const revision = `${branch}~${i}`
+
+      const commitDataOutput = await this.internalCommands.executeCommand(
+        AvailableCommands.GIT_GET_COMMIT_MESSAGES,
+        this.dvcRoot,
+        revision
+      )
+
+      const { hash, ...commitData } = getCommitDataFromOutput(commitDataOutput)
+
+      if (hashes[hash]) {
+        hashes[hash].branches.push(branch)
+      } else {
+        hashes[hash] = {
+          branches: [branch],
+          ...commitData
+        }
+      }
+
+      flags.push(ExperimentFlag.REV, revision)
+    }
+  }
+
+  private addDetailsToRevision(
+    out: ExpState & {
+      experiments?: ExpRange[] | null | undefined
+    },
+    data: ExpShowOutput,
+    hashes: Record<string, HashInfo>
+  ) {
+    const revision = hashes[out.rev]
+    if (revision) {
```

Comment:
This `if` was added to get tests to pass at this stage. Might be a good thing to remove it once fixtures are in place correctly.